src/lib/InstantSearch.ts

```diff
@@ -256,21 +281,25 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
     };
 
     if (this._searchFunction) {
+      // this client isn't used to actually search, but required for the helper
+      // to not throw errors
+      const fakeClient = ({
```

Comment:
It would be less of a lie to type it as `Partial<Client>` and to cast it as `Client` line 294.

```ts
const fakeClient = {
  search: () => new Promise(noop),
} as Partial<Client>;
```